sycl/source/detail/image_accessor_util.cpp

```diff
@@ -29,12 +29,67 @@ cl_int4 getPixelCoordNearestFiltMode(cl_float4 Coord_uvw,
   cl_int height = ImgRange[1];
   cl_int depth = ImgRange[2];
   switch (SmplAddrMode) {
-  case addressing_mode::mirrored_repeat:
-    // TODO: Add the computations.
-    break;
-  case addressing_mode::repeat:
-    // TODO: Add the computations.
-    break;
+  case addressing_mode::mirrored_repeat: {
+    cl_float u1, v1, w1;
+    u1 = 2.0f * cl::sycl::rint(0.5f * u);
+    v1 = 2.0f * cl::sycl::rint(0.5f * v);
+    w1 = 2.0f * cl::sycl::rint(0.5f * w);
+
+    u1 = cl::sycl::fabs(u-u1);
+    v1 = cl::sycl::fabs(v-v1);
+    w1 = cl::sycl::fabs(w-w1);
+
+    u1 = u1 * width;
+    v1 = v1 * height;
+    w1 = w1 * depth;
+
+    i = (int)cl::sycl::floor(u1);
+    j = (int)cl::sycl::floor(v1);
+    k = (int)cl::sycl::floor(w1);
+
+    i = cl::sycl::min(i, (width-1));
+    j = cl::sycl::min(j, (height-1));
+    k = cl::sycl::min(k, (depth-1));
+    // Eg:
+    // u,v,w = {2.3,1.7,0.5} // normalized coordinates.
+    // w,h,d = {9,9,9}
+    // u1=2*rint(1.15)=2
+    // v1=2*rint(0.85)=2
+    // w1=2*rint(0.5)=0
+    // u1=fabs(2.3-2)=.3
+    // v1=fabs(1.7-2)=.3
+    // w1=fabs(0.5-0)=.5
+    // u1=0.3*9=2.7
+    // v1=0.3*9=2.7
+    // w1=0.5*9=4.5
+    // i,j,k = {2,2,4}
+
+  } break;
+  case addressing_mode::repeat: {
+    cl_float u1 = (u - cl::sycl::floor(u)) * width;
```

Comment:
the same sentences, instead of an example it would be better to have a unit test test for a separate implementation of this case.

Reply:
edited these too.